experimental/add_to_app/android_fullscreen/app/src/main/java/dev/flutter/example/androidfullscreen/MainActivity.kt

```diff
@@ -6,18 +6,71 @@ package dev.flutter.example.androidfullscreen
 
 import android.os.Bundle
 import android.widget.Button
+import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.embedding.engine.FlutterEngineCache
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.view.FlutterMain
 
 class MainActivity : AppCompatActivity() {
 
+    private val ENGINE_ID = "1"
+
+    private var channel : MethodChannel? = null
+    private var count = 0
+    private var counterLabel : TextView? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        FlutterMain.ensureInitializationComplete(applicationContext, null)
```

Comment:
@matthew-carroll 

This line isn't something we talked about, but I found it necessary to get past this check:

https://github.com/flutter/engine/blob/master/shell/platform/android/flutter_main.cc#L50

Is this the correct approach, or have I hacked my way around a problem the app shouldn't have?

Reply:
For now, you'll need:

```
FlutterMain.startInitialization(...);
FlutterMain.ensureInitializationComplete(...);
```

I'm working on a PR right now that @xster started, which will move those calls inside of `FlutterEngine` so that app devs no longer have to worry about it.

Reply:
SGTM. Thanks.